File: Web apps/FIR Example/MyFir.js
/**
 * @brief Implementation of finite impulse response (FIR) filter.
 */
class MyFir {
	
	/**
	 * @brief Initialization of FIR filter algorithm. 
	 * @param ffc Array of FIR filter feedforward coefficients.
	 * @param st Array of FIR filter input initial state.
	 *        Must be the same size as coefficients array.
	 */
	 constructor(ffc, st) {
		this.feedforward_coefficients = ffc;
		this.state = st;
	 }
	 
	/**
	 * @brief Execute one step of the FIR filter algorithm.
	 * @param x Input signal.
	 * @retval Output [filtered] signal.
	 */
	Execute(x) {
		// update state
		this.state.unshift(x);
		this.state.pop();
		// compute output
		let xf = 0.0;
		for (let i = 0; i < this.state.length; i++) {
			xf += this.feedforward_coefficients[i]*this.state[i];
		}
		return xf;
	}
}

File: Desktop/FIR Design Example/python/MyFir/MyFir.py

class MyFir:

  # @brief Initialization of FIR filter algorithm. 
  # @param ffc : List of FIR filter feedforward coefficients.
  # @param st  : List of FIR filter input initial state.
  #              Must be the same size as coefficients list.
  def __init__(self, fft, st):
    self.feedforward_coefficients = fft; ##!< List of FIR filter feedforward coefficients
    self.state = st;                     ##!< List of FIR filter state values
  
  # @brief Execute one step of the FIR filter algorithm.
  # @param x : Input signal.
  # @return Output [filtered] signal.
  def Execute(self, x):
    # update state
    self.state.pop()
    self.state.insert(0, x)
    # compute output 
    xf = 0.0
    for i in range(len(self.state)):
      xf += self.feedforward_coefficients[i]*self.state[i]
    return xf
